libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -1108,8 +1160,30 @@ static CK_RV prvSaveDerKeyToPal( mbedtls_pk_context * pxMbedContext,
     uint32_t ulActualKeyLength = 0;
     int32_t lCompare = 0;
     CK_OBJECT_HANDLE xPalHandle = CK_INVALID_HANDLE;
+    uint32_t ulDerBufSize = 0;
+
+    if( ( xKeyType == CKK_EC ) && ( xIsPrivate == ( CK_BBOOL ) CK_TRUE ) )
+    {
+        ulDerBufSize = EC_MAX_LENGTH_KEY;
+    }
+    else if( ( xKeyType == CKK_EC ) && ( xIsPrivate == ( CK_BBOOL ) CK_FALSE ) )
+    {
+        ulDerBufSize = EC_MAX_PUBLIC_KEY_SIZE;
+    }
+    else if( ( xKeyType == CKK_RSA ) && ( xIsPrivate == ( CK_BBOOL ) CK_TRUE ) )
+    {
+        ulDerBufSize = MAX_LENGTH_KEY;
+    }
+    else if( ( xKeyType == CKK_RSA ) && ( xIsPrivate == ( CK_BBOOL ) CK_FALSE ) )
+    {
+        ulDerBufSize = MAX_PUBLIC_KEY_SIZE;
+    }
+    else
+    {
+        xResult = CKR_ARGUMENTS_BAD;
+    }
 
-    pxDerKey = pvPortMalloc( MAX_PUBLIC_KEY_SIZE );
+    pxDerKey = pvPortMalloc( ulDerBufSize );
```

Comment:
Please check for `if( xResult == CKR_OK )` before calling pcPortMalloc.

Reply:
Good catch, thank you. Updated.